tools/config/seed.config.ts

```diff
@@ -379,28 +379,52 @@ export class SeedConfig {
   ];
 
   /**
-   * The BrowserSync configuration of the application.
-   * The default open behavior is to open the browser, 
-   * To prevent the browser from opening
-   * `--b`  flag when running `npm start` (tested with serve.dev)
-   * example `npm start -- --b`
-   * @type {any}
-   */
-  BROWSER_SYNC_CONFIG: any = {
-    middleware: [require('connect-history-api-fallback')({ index: `${this.APP_BASE}index.html` })],
-    port: this.PORT,
-    startPath: this.APP_BASE,
-    open: argv['b'] ? false : true,
-    server: {
-      baseDir: `${this.DIST_DIR}/empty/`,
-      routes: {
-        [`${this.APP_BASE}${this.APP_DEST}`]: this.APP_DEST,
-        [`${this.APP_BASE}node_modules`]: 'node_modules',
-        [`${this.APP_BASE.replace(/\/$/, '')}`]: this.APP_DEST
+   * Configurations for NPM module configurations. Add to or override in project.config.ts.
+   * If you like, use the mergeObject() method to assist with this.
+   */
+  PLUGIN_CONFIGS: any = {
+    /**
+     * The BrowserSync configuration of the application.
+     * The default open behavior is to open the browser, 
+     * To prevent the browser from opening
+     * `--b`  flag when running `npm start` (tested with serve.dev)
+     * example `npm start -- --b`
+     * @type {any}
+     */
+    'browser-sync': {
+      middleware: [require('connect-history-api-fallback')({ index: `${this.APP_BASE}index.html` })],
+      port: this.PORT,
+      startPath: this.APP_BASE,
+      open: argv['b'] ? false : true,
+      server: {
+        baseDir: `${this.DIST_DIR}/empty/`,
+        routes: {
+          [`${this.APP_BASE}${this.APP_DEST}`]: this.APP_DEST,
+          [`${this.APP_BASE}node_modules`]: 'node_modules',
+          [`${this.APP_BASE.replace(/\/$/, '')}`]: this.APP_DEST
+        }
       }
     }
   };
 
+  /**
+   * Recursively merge source onto target.
+   */
+  mergeObject( target: any, source: any ) {
+    var deepExtend = require('deep-extend');
```

Comment:
I'm not big fan of `require` here. I'd prefer a definition file and import the module using the `import` syntax where needed. Don't you think so ?